Pear/Profile/ProfileViewController.swift

```diff
@@ -9,19 +9,34 @@ import GoogleSignIn
 import Kingfisher
 import UIKit
 
+enum ProfileViewing {
+    case currentUser
+    case otherUser
+}
+
 class ProfileViewController: UIViewController {
 
     // MARK: - Private View Vars
     private let backButton = UIButton()
     private var currentUser: UserV2
-    private var otherUser: UserV2?
+    private let editButton = UIButton()
+    private var loadedUser: UserV2?
     private var profileSections = [ProfileSectionType]()
     private let profileTableView = UITableView(frame: .zero, style: .plain)
+    private let viewType: ProfileViewing
 
-    init(user: UserV2, otherUserId: Int) {
+    init(user: UserV2, viewType: ProfileViewing, otherUserId: Int?) {
         self.currentUser = user
+        self.viewType = viewType
         super.init(nibName: nil, bundle: nil)
-        getOtherUser(otherUserId: otherUserId)
+        switch viewType {
+            case .currentUser:
+                loadCurrentUser()
+            case .otherUser:
+                if let otherUserId = otherUserId {
+                    getOtherUser(otherUserId: otherUserId)
+                }
```

Comment:
if `otherUserId` is wrapped in `.otherUser`, you could avoid the possibility of having an `.otherUser` `viewType` with no `otherUserId`.